code/modules/black_market/black_market_beacon.dm

```diff
@@ -0,0 +1,40 @@
+var/list/black_market_beacons = list()
+
+/obj/item/device/black_market_beacon
+	name = "black market beacon"
+	desc = "You're really not sure how you managed to grab one of these without it melting."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "implant_evil"
+	w_class = W_CLASS_TINY
+	item_state = ""
+	throw_speed = 4
+	throw_range = 20
+	var/datum/black_market_player_item/market_listing
+	var/atom/attached_item
+
+/obj/item/device/black_market_beacon/proc/attach_to(var/atom/target, var/datum/black_market_player_item/listing)
```

Comment:
wait what holy shit you can attach beacons to **any atom**.

this means you can literally buy people out of mechs.

also expect the obligatory listing to teleport people out of danger areas. "; help my body on roid buy my listing"